### PR TITLE
Add support for Terraform 0.12

### PR DESCRIPTION
**Pull request by remilapeyre**

### CODE REVIEW

File: consul/resource_consul_prepared_query_test.go [hunk 1]
```diff
@@ -163,6 +163,10 @@ func testAccConsulPreparedQueryNearestN(t *testing.T) func() {
 	}
 }
 
+func testAccCheckConsulPreparedQueryAttrValue(attr, val string) resource.TestCheckFunc {
```
Comment (bflad) [inline consul/resource_consul_prepared_query_test.go]: This new function appears to be unused.

File: consul/resource_consul_keys.go [hunk 2]
```diff
@@ -19,6 +19,13 @@ func resourceConsulKeys() *schema.Resource {
 		SchemaVersion: 1,
 		MigrateState:  resourceConsulKeysMigrateState,
 
+		CustomizeDiff: func(d *schema.ResourceDiff, _ interface{}) error {
```
Comment (bflad) [inline consul/resource_consul_keys.go]: Was this change intended for this pull request?

Comment (remilapeyre) [reply consul/resource_consul_keys.go]: Yes, this is part of ff2785a7 and related to https://github.com/hashicorp/terraform/issues/21070.

I think (but I'm not sure) that it is because Terraform 0.12 now expect the result of the apply to match the plan so updating the SDK highlighted a bug that was previously silent.

**Comment by remilapeyre** [comment]
Hi @bflad, thanks for the review. It should be good to go now.
